frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -485,6 +490,11 @@ function ReaderBookmark:updateBookmark(item)
     for i=1, #self.bookmarks do
         if item.datetime == self.bookmarks[i].datetime and item.page == self.bookmarks[i].page then
             local page = self.ui.document:getPageFromXPointer(item.updated_highlight.pos0)
+            local flow = self.ui.document:getPageFlow(page)
+            page = self.ui.document:getPageNumberInFlow(page)
+            if flow > 0 then
+                page = T("[%1]%2", page, flow)
+            end
```

Comment:
I'd be happier if that kind of things would be written:
```lua
            local page = self.ui.document:getPageFromXPointer(item.updated_highlight.pos0)
            local flow = self.ui.document:getPageFlow(page) -- 0 when no (or not hidden) non-linear flows
            if flow > 0 then
                page = self.ui.document:getPageNumberInFlow(page)
                page = T("[%1]%2", page, flow)
            end
```
or even:
```lua
            local page = self.ui.document:getPageFromXPointer(item.updated_highlight.pos0)
            if self.ui.document:hasHiddenFlows() then
                local flow = self.ui.document:getPageFlow(page) -- 0 when no (or not hidden) non-linear flows
                if flow > 0 then
                    page = self.ui.document:getPageNumberInFlow(page)
                    page = T("[%1]%2", page, flow)
                end
            end
```
no matter the additional function call.